confidential-data-hub/image/src/annotation_packet/v2.rs

```diff
@@ -61,36 +62,27 @@ fn default_provider() -> String {
 }
 
 impl TryInto<super::v1::AnnotationPacket> for AnnotationPacketV2 {
-    type Error = Error;
+    type Error = anyhow::Error;
 
     fn try_into(self) -> std::result::Result<super::v1::AnnotationPacket, Self::Error> {
         if self.version != DEFAULT_VERSION {
-            return Err(Error::ConvertAnnotationPacketFailed(format!(
-                "`version` must be {DEFAULT_VERSION}."
-            )));
+            bail!("`version` must be {DEFAULT_VERSION}.");
```

Comment:
should we resort to anyhow in library code? If we want to reduce verbosity, we could move the strings to a thiserror enum:

```
return Err(Error::PackageVersionMismatch(self.version));
...
return Err(Error::UnknownProvider(self.provider));
...
```

Reply:
Oh. This is all about transform `PacketV1` to `PacketV2`, let me change the error information of `ConvertAnnotationPacketFailed`.